### ISSUE TITLE
Does the promise return from speak() actually work?

### ISSUE THREAD

**Issue by elkay** [RaSan147/pixi-live2d-display#3]
Unless I'm doing something wrong, I see the promise as false right away as soon as playback starts and never fulfilled.

[[Prototype]]: Promise
[[PromiseState]] : "fulfilled"
[[PromiseResult]] : false

Or maybe I'm just not doing it right. Not new to promises, which is why I'm a bit confused why I can't get this going.

**Comment by RaSan147** [comment]
Can you kindly send the code?
I'm still debugging, after refactoring (as per guanzz asked for changes)
there is some instability 😥

**Comment by elkay** [comment]
No worries! Appreciate the quick response. I was also reviewing your code to see if I saw something out of alignment there.

Also want to make sure my use case is what the intent of this was. I have multiple audio files I want to play (speak) sequentially without interrupting each other. I was planning to queue them up and use the promise, while it's true I thought it was playing and false means it was done playing. Is that not how this was designed? Regardless, I never saw "true" come out.

Was basing it on an interpretation of this:

     * @returns Promise that resolves with true if the sound is playing, false if it's not

Thanks!

**Comment by RaSan147** [comment]
As far i remember, based on the module design, it probably returns as soon
as the task begins

I recommend creating an audio element to check the duration of the audio

Here is the psudo code(writing on email)
async function play_audios(){
    Var Files = [list of audio links]
    For link in Files{
       Var audio_ele =new Audio()
       Audio_ele.src = link
       Var dur = audio_ele.duration
       Await Model.speak(link)
       Await sleep(dur) # make async sleep function (try google)

    }
}

**Comment by elkay** [comment]
Yeah I did more digging and looked like at least the intent is for the return once audio starts. Won't work for my needs, however I did notice I have access the the motionManager and it has a currently playing audio property that I have access to. I can poll that on a short interval to accomplish the same goal, so I'll likely just go with that for now.

**Comment by elkay** [comment]
Btw, that is what I did try..

 Await Model.speak(link)

It didn't wait at all. Moved on as soon as audio started playing.

**Comment by RaSan147** [comment]
Sounds like a plan! I could return the current audio, but it was based on
the design of the original author, sorry can't help much on that.

**Comment by RaSan147** [comment]
Where?! Can you share the part of the code please, I found some issues on
motion with audio on my end too.

**Comment by elkay** [comment]
await this.pixiModel.speak(audioUrl, { volume: 1, expression: null, resetExpression: true, crossOrigin: 'anonymous' });
          console.log('DONE!!!');
          
          this.pixiModel.speak(audioUrl, { volume: 1, expression: null, resetExpression: true, crossOrigin: 'anonymous' }).then(() => {
            console.log('DONE!!!');
          });*/
          
"DONE!!!" immediately to console with both examples.

**Comment by RaSan147** [comment]
Just as i said, it fires as soon as the audio starts (and its kinda
instantly) so await doesn't take time and behaves asap. You can check the
return value... Did both time the audio played?

**Comment by elkay** [comment]
Yeah the audio plays fine. I'm just going to go the other way about it via motionManager exposed audio. Thanks!

**Comment by RaSan147** [comment]
Thanks for knocking. This is my 1st working on TS, so lemme know if there's any unstablity.
